### PR TITLE
[Merged by Bors] - feat: port Order.SuccPred.IntervalSucc

### PR DESCRIPTION
**Pull request by mo271**

### CODE REVIEW

File: Mathlib/Order/SuccPred/IntervalSucc.lean [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+import Mathlib.Data.Set.Pairwise
+import Mathlib.Order.SuccPred.Basic
+
+/-!
+# Intervals `Ixx (f x) (f (order.succ x))`
+
+In this file we prove
+
+* `monotone.bUnion_Ico_Ioc_map_succ`: if `α` is a linear archimedean succ order and `β` is a linear
+  order, then for any monotone function `f` and `m n : α`, the union of intervals
```
Comment (dupuisf) [inline Mathlib/Order/SuccPred/IntervalSucc.lean:20]:
```suggestion
* `Monotone.bunionᵢ_Ico_Ioc_map_succ`: if `α` is a linear archimedean succ order and `β` is a linear
  order, then for any monotone function `f` and `m n : α`, the union of intervals
```

File: Mathlib/Order/SuccPred/IntervalSucc.lean [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+  `set.Ioc (f i) (f (order.succ i))`, `m ≤ i < n`, is equal to `set.Ioc (f m) (f n)`;
+
+* `monotone.pairwise_disjoint_on_Ioc_succ`: if `α` is a linear succ order, `β` is a preorder, and
+  `f : α → β` is a monotone function, then the intervals `set.Ioc (f n) (f (order.succ n))` are
```
Comment (dupuisf) [inline Mathlib/Order/SuccPred/IntervalSucc.lean:24]:
```suggestion
* `Monotone.pairwise_disjoint_on_Ioc_succ`: if `α` is a linear succ order, `β` is a preorder, and
  `f : α → β` is a monotone function, then the intervals `Set.Ioc (f n) (f (Order.succ n))` are
```

File: Mathlib/Order/SuccPred/IntervalSucc.lean [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+/-!
+# Intervals `Ixx (f x) (f (order.succ x))`
+
+In this file we prove
+
+* `monotone.bUnion_Ico_Ioc_map_succ`: if `α` is a linear archimedean succ order and `β` is a linear
+  order, then for any monotone function `f` and `m n : α`, the union of intervals
+  `set.Ioc (f i) (f (order.succ i))`, `m ≤ i < n`, is equal to `set.Ioc (f m) (f n)`;
+
```
Comment (dupuisf) [inline Mathlib/Order/SuccPred/IntervalSucc.lean:22]:
```suggestion
  `Set.Ioc (f i) (f (Order.succ i))`, `m ≤ i < n`, is equal to `Set.Ioc (f m) (f n)`;

```

**Review by dupuisf — CHANGES_REQUESTED** [review]
A lot of lemma names need to be adjusted according to mathlib4's new naming scheme -- you can have a look at the [porting wiki](https://github.com/leanprover-community/mathlib4/wiki/Porting-wiki) in the section "Naming convention" to see how this works.

**Comment by mo271** [comment]
> A lot of lemma names need to be adjusted according to mathlib4's new naming scheme -- you can have a look at the [porting wiki](https://github.com/leanprover-community/mathlib4/wiki/Porting-wiki) in the section "Naming convention" to see how this works.

I corrected the bUnion thing and also the many names in the comments. Is there more, which I didn't see?

**Review by dupuisf — COMMENTED** [review]
Thanks!

bors r+

**Pull request closed without merging**